manifests/plugin/table.pp

```diff
@@ -0,0 +1,24 @@
+# https://collectd.org/wiki/index.php/Chains
+class collectd::plugin::table (
+  Enum['present', 'absent'] $ensure    = 'present',
+  Integer $order                       = 10,
+  Hash[String, Struct[{
```

Comment:
Thanks for this awesome and detailed datatype! Can you please abstract it into its own datatype in https://github.com/voxpupuli/puppet-collectd/tree/master/types ?

Reply:
Done, awesome I was not aware we can declare type there :)